test/abstract_syntax_tree.coffee

```diff
@@ -10,207 +10,595 @@ getExpressions = (code) ->
+    ]
+  # TODO: `'fn'` identifier is missing from AST.
 
 test "AST as expected for StatementLiteral node", ->
-  expression = getExpression 'break'
-  eq expression.type, 'StatementLiteral'
-  eq expression.value, 'break'
+  testExpression 'break',
+    type: 'StatementLiteral'
+    value: 'break'
 
 test "AST as expected for ThisLiteral node", ->
-  expression = getExpression 'this'
-  eq expression.type, 'ThisLiteral'
-  eq expression.value, 'this'
+  testExpression 'this',
+    type: 'ThisLiteral'
+    value: 'this'
 
-  expression = getExpression '@'
-  eq expression.type, 'ThisLiteral'
-  eq expression.value, 'this'
-  # TODO: Check for raw value of '@'.
+  testExpression '@',
+    type: 'ThisLiteral'
+    value: 'this'
+    # originalValue: '@'
+  # TODO: `@` literal is not yet preserved in ast generation.
 
 test "AST as expected for UndefinedLiteral node", ->
-  expression = getExpression 'undefined'
-  eq expression.type, 'UndefinedLiteral'
-  eq expression.value, 'undefined'
+  testExpression 'undefined',
+    type: 'UndefinedLiteral'
+    value: 'undefined'
 
 test "AST as expected for NullLiteral node", ->
-  expression = getExpression 'null'
-  eq expression.type, 'NullLiteral'
-  eq expression.value, 'null'
+  testExpression 'null',
+    type: 'NullLiteral'
+    value: 'null'
 
 test "AST as expected for BooleanLiteral node", ->
-  expression = getExpression 'true'
-  eq expression.type, 'BooleanLiteral'
-  eq expression.value, 'true'
+  testExpression 'true',
+    type: 'BooleanLiteral'
+    value: 'true'
 
-  expression = getExpression 'off'
-  eq expression.type, 'BooleanLiteral'
-  eq expression.value, 'false'
-  # TODO: Check for raw value of 'off'.
+  testExpression 'off',
+    type: 'BooleanLiteral'
+    value: 'false'
+    originalValue: 'off'
+
+  testExpression 'yes',
+    type: 'BooleanLiteral'
+    value: 'true'
+    originalValue: 'yes'
 
 test "AST as expected for Return node", ->
-  expression = getExpression 'return no'
-  eq expression.type, 'Return'
-  eq expression.expression.type, 'BooleanLiteral'
+  testExpression 'return no',
+    type: 'Return'
+    expression: type: 'BooleanLiteral'
 
 test "AST as expected for YieldReturn node", ->
-  expression = getExpression 'yield return ->'
-  eq expression.type, 'YieldReturn'
-  eq expression.expression.type, 'Code'
+  testExpression 'yield return ->',
+    type: 'YieldReturn'
+    expression: type: 'Code'
 
 test "AST as expected for AwaitReturn node", ->
-  expression = getExpression 'await return ->'
-  eq expression.type, 'AwaitReturn'
-  eq expression.expression.type, 'Code'
+  testExpression 'await return ->',
+    type: 'AwaitReturn'
+    expression: type: 'Code'
 
 test "AST as expected for Value node", ->
-  expression = getExpression 'for i in [] then i'
-  value = expression.body
-  eq value.type, 'Value'
-  # eq value.value, 'i'
+  testExpression 'for i in [] then i',
+    body:
+      type: 'Value'
+      isDefaultValue: no
+      base: value: 'i'
```

Comment:
In general can we avoid implicit objects as object values? It seems like a recipe for trouble later.
```coffee
base:
  value: 'i'
```

Reply:
Well, we could. But not only would it make the file much longer, it would also make it less readable IMO.

Reply:
Unless I’m mistaken, it’s not the coding style of the rest of the codebase. We try to keep things consistent.